source/ranges.tex

```diff
@@ -634,12 +634,11 @@
   \end{itemize}
 
 \item
-  Otherwise, \tcode{\placeholdernc{make-unsigned-like}(ranges::end(E) - ranges::begin(E))}\iref{range.subrange}
-  if it is a valid expression and
-  the types \tcode{I} and \tcode{S} of \tcode{ranges::begin(E)} and
-  \tcode{ranges::end(E)} (respectively) model both
-  \tcode{\libconcept{sized_sentinel_for}<S, I>}\iref{iterator.concept.sizedsentinel} and
-  \tcode{\libconcept{forward_iterator}<I>}.
+  Otherwise, \tcode{\placeholdernc{make-unsigned-like}(ranges::end(E) - ranges::begin(E))}\iref{ranges.syn}
+  if it is a valid expression, and
+  \tcode{T} models \tcode{forward_range}\iref{range.refinements}, and
+  \tcode{iterator_t<T>} and \tcode{sentinel_t<T>}\iref{ranges.syn} model
```

Comment:
How editorially-obvious is it that `iterator_t&lt;T>` and the type of `ranges::begin(E)` are actually the same type? For example, could `ranges::begin(E)` return an lvalue, which would make `iterator_t&lt;T>` a reference? (i think the answer is "no", because weakly_incrementable requires default-constructible. And the answer is also "no" for the similar question regarding `ranges::end(E)`, because semiregular also requires default-constructible. But it takes a few hops to get there.)

Reply:
No, they cannot return references. They either return a pointer, or _decay-copy_`(something).